data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ClientBuilderFactory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import java.time.Duration;
+
+/**
+ * A Builder Factory for creating and fetching the <code>S3Client</code> and <code>S3AsyncClient</code>.
+ */
+public class S3ClientBuilderFactory {
```

Comment:
Nit: Builder and Factory are redundant unless you want a Factor that creates builders